### PR TITLE
Added screenreader support

### LINKED ISSUE #205: Current note editor is not accessible to screenreaders
Unfortunately, the editor to compose note contents is not accessible using screen readers at present. From the behavior I'm seeing I think it's either Ace or CodeMirror, which unfortunately don't work well with assistive technology.
Please consider alternatives like Monaco, or an alternative where screen reader users are able to use a standard <textarea> or the contenteditable attribute in order to fix this, as at present it is a showstopper and prevents people using screenreaders such as myself to use Archivy the way it was meant to be used.
Temporary workaround is to write note contents somewhere else and copy-paste it into the field, which works, but is far from ideal.
Awesome concept for the rest, and thanks a lot for reading at least! :)

### PR DESCRIPTION
**Pull request by ddoyaguez**
Added screenreader support that fixes #205

### CODE REVIEW

**Comment by Uzay-G** [comment]
Thanks for opening this PR, it looks good :). I'll check it out :+1:

**Review by srevinsaju — APPROVED** [review]

File: archivy/templates/dataobjs/show.html [hunk 1]
```diff
@@ -107,6 +114,46 @@ <h2 id="post-title">{{ dataobj['title'] }}</h2>
           oldContent = document.getElementById("content"), editorDiv = document.querySelector(".EasyMDEContainer");
           editorDiv.classList.add("hidden");
           document.getElementById("post-title").scrollIntoView(true);
+
+          /* Screenreader mode checkbox */
+          function toggleScreenReaderMode(cbox, editor)
+          {
+            return function ()
```
Comment (Uzay-G) [inline archivy/templates/dataobjs/show.html:121]: Do we need an inner function here, or could we just directly put the if/else?

(Just curious, I might be making a mistake :smile: )

File: archivy/templates/dataobjs/show.html [hunk 2]
```diff
@@ -107,6 +114,46 @@ <h2 id="post-title">{{ dataobj['title'] }}</h2>
...
+            {
+              if (cbox.checked) {
+                editor.toTextArea();
+                editor = null;
+              } else {
+                location.reload();
```
Comment (Uzay-G) [inline archivy/templates/dataobjs/show.html:127]: Could you explain why the page is reloaded here? I understand for when the button is checked, but don't understand the other side.

File: archivy/templates/dataobjs/show.html (hunk 1 above)
Comment (ddoyaguez) [reply archivy/templates/dataobjs/show.html:121]: Yes, we need the inner function in order to pass the editor object to the function. I think we could remove the cbox parameter and use "this" to reference it from the inside, but I think it is clearer this way.

File: archivy/templates/dataobjs/show.html (hunk 2 above)
Comment (ddoyaguez) [reply archivy/templates/dataobjs/show.html:127]: Ok so, I needed a fast way to revert to the previous state (altough I don't think any screenreader user would deactivate the checkbox , maybe there is some circunstance under which the user might want to revert to default state), and I think it is a more direct, effective and clean (in terms of code readability) way of doing it than reinstantiating easyMDE :)

**Comment by Uzay-G** [comment]
Ok, seems good :)

**Pull request merged**